aten/src/ATen/cpu/vec256/vec256_double.h

```diff
@@ -307,11 +307,16 @@ inline void convert(const double* src, double* dst, int64_t n) {
   }
 }
 
-#ifdef __AVX2__
+#if defined(__AVX2__) && defined(__FMA__)
 template <>
 Vec256<double> inline fmadd(const Vec256<double>& a, const Vec256<double>& b, const Vec256<double>& c) {
   return _mm256_fmadd_pd(a, b, c);
 }
+#elif defined(__AVX__)
+template <>
+Vec256<double> inline fmadd(const Vec256<double>& a, const Vec256<double>& b, const Vec256<double>& c) {
+  return a * b + c;
+}
```

Comment:
I don't think this is a good idea, it's changing numerics based on the available vector extension, and that is not clear to a user of vec256

Can we just do this at the callsites?

Reply:
Sure: separate that part into a new PR: https://github.com/pytorch/pytorch/pull/29154